esmvaltool/diag_scripts/primavera/energy_budget.py

```diff
@@ -0,0 +1,249 @@
+                                                data['hfss'][dataset] -
+                                                data['hfls'][dataset])
+            long_name = 'Radiation Adsorbed Surface'
+            data['rad_ads_surface'][dataset].long_name = long_name
+
+            data['rad_net_toa'][dataset] = (data['rsdt'][dataset] -
+                                            data['rsut'][dataset] -
+                                            data['rlut'][dataset])
+            long_name = 'Radiation Net TOA'
+            data['rad_net_toa'][dataset].long_name = long_name
+
+            data['bowen_ratio'][dataset] = (data['hfss'][dataset] /
+                                            data['hfls'][dataset])
+            data['bowen_ratio'][dataset].long_name = 'Bowen Ratio'
+
+            cubes = iris.cube.CubeList([
+                data['rsdt'][dataset],
+                data['rsut'][dataset],
+                data['rsds'][dataset],
+                data['rsns'][dataset],
+                data['rlut'][dataset],
+                data['rlds'][dataset],
+                data['rlns'][dataset],
+                data['hfss'][dataset],
+                data['hfls'][dataset],
+                data['up_sw_rfl_surf'][dataset],
+                data['sw_rfl_clouds'][dataset],
+                data['sw_abs_atm'][dataset],
+                data['up_lw_emit_surf'][dataset],
+                data['net_surf_rad'][dataset],
+                data['rad_ads_surface'][dataset],
+                data['rad_net_toa'][dataset],
+                data['bowen_ratio'][dataset]
+            ])
+            self.save(dataset, cubes)
+        if self.template:
+            self.plot(data)
+
+    def load(self):
+        data = {}
+        data['rsdt'] = {}
+        data['rsut'] = {}
+        data['rsds'] = {}
+        data['rsns'] = {}
+        data['rlut'] = {}
+        data['rlds'] = {}
+        data['rlns'] = {}
+        data['hfss'] = {}
+        data['hfls'] = {}
+        data['up_sw_rfl_surf'] = {}
+        data['sw_rfl_clouds'] = {}
+        data['sw_abs_atm'] = {}
+        data['up_lw_emit_surf'] = {}
+        data['net_surf_rad'] = {}
+        data['rad_ads_surface'] = {}
+        data['rad_net_toa'] = {}
+        data['bowen_ratio'] = {}
+        for filename in self.filenames:
+            dataset = self.filenames.get_info(n.DATASET, filename)
+            short_name = self.filenames.get_info(n.SHORT_NAME, filename)
+            self.project = self.filenames.get_info('project', filename)
+            self.exp = self.filenames.get_info('exp', filename)
+            self.start = self.filenames.get_info('start_year', filename)
+            self.end = self.filenames.get_info('end_year', filename)
+            cube = iris.load_cube(filename)
+            data[short_name][dataset] = cube
+
+        return data
+
+    def save(self, dataset, cubes):
+        file_name = ('{project}_{dataset}_{exp}_{script}_variables'
```

Comment:
Maybe it would be better to use the `alias` key here and in other places, because it looks like things will go wrong if someone would put more than one ensemble member from the same model in the recipe